internal/build/job/manager.go

```diff
@@ -83,6 +87,20 @@ func (jbm *jobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1b
 		return build.Result{Status: build.StatusCreated, Requeue: true}, nil
 	}
 
+	changed, err := jbm.isJobChanged(mod, buildConfig, targetKernel, m.ContainerImage, pushImage, job)
```

Comment:
Should we always generate the new job and compare it to the existing one? That would simplify `isJobChanged` and reduce the number of parameters it receives.

Reply:
Why would we want to generate a job even if there is no need for it?

Reply:
yes, good suggestion it will simplify everything, fixed

Reply:
I though that @qbarrand meant to always create a new job in k8s and then compare the 2 running jobs.
Only after the changes I understood what you meant :)

Reply:
I am renaming MakeJob -> MakeJobTemplate  to avoid future confusion